cmd/plan.go

```diff
@@ -162,7 +169,72 @@ func planInstall(rv cran.RVersion) (*cran.PkgDb, gpsr.InstallPlan) {
 	for pkg := range ip.DepDb {
 		pkgs = append(pkgs, pkg)
 	}
+
 	log.Infoln("total packages required:", len(ip.StartingPackages)+len(ip.DepDb))
 	log.Infoln("resolution time", time.Since(startTime))
 	return cdb, ip
 }
+
+
+type InstalledPackage struct {
+	Name string
+	Version string
+	Repo string
+}
+
+func GetPriorInstalledPackages(fileSystem afero.Fs, libraryPath string) map[string]InstalledPackage {
+
+	installed := make(map[string]InstalledPackage)
+
+	installedLibrary, err := fileSystem.Open(libraryPath)
+
+	if err != nil {
+		//panic?
+		return installed
+	}
+
+	installedPkgs, _ := installedLibrary.Readdir(0)
+
+	for _, f := range installedPkgs {
+		descriptionFilePath := filepath.Join(libraryPath, f.Name(), "DESCRIPTION")
```

Comment:
this whole chunk of code should be broken out to a separate set of function(s). For example readDescription (returns a byte slice or string slice). 

getPackageMetadata() PkgMetadata 

where the PkgMetadata struct would have like Version/Name/other?

these could then be tested and managed independently and probably should be in a separate package entirely